src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/SetPosition.java

```diff
@@ -29,9 +32,18 @@ public SetPosition(ConnectionImpl connection) {
 		}
 		//PE sends position that intersects blocks bounding boxes in some cases
 		//Server doesn't accept such movements and will send a set position, but we ignore it unless it is above leniency
-		if (cache.getMovementCache().isPEPositionAboveLeniency()) {
+		if (movecache.isPEPositionAboveLeniency()) {
 			packets.add(create(cache.getWatchedEntityCache().getSelfPlayer(), x, y + 0.01, z, pitch, yaw, ANIMATION_MODE_TELEPORT));
 		}
+		movecache.setPEClientPosition(x, y, z);
+		if (version.isAfterOrEq(ProtocolVersion.MINECRAFT_PE_1_8)) {
+			ClientBoundPacketData networkChunkUpdate = ClientBoundPacketData.create(PEPacketIDs.NETWORK_CHUNK_PUBLISHER_UPDATE_PACKET);
```

Comment:
Create a static function like the one https://github.com/ProtocolSupport/ProtocolSupport/blob/6efbff355557502b0c33db5fd11fedc46e5c3ae1/src/protocolsupport/protocol/packet/middleimpl/clientbound/play/v_pe/Chunk.java#L75 inside the chunk packet. Then use that code here so we share the packet structure.

Reply:
this is not exactly what I meant. I meant that since this is exactly the same code as you have in Chunk it is better to make a public static ClientBoundPacketData that takes all arguments needed. So, inside Chunk.java create:
```java
public static ClientBoundPacketData createChunkPublisherUpdate(int x, int y, int z, int view) {
	 		    ClientBoundPacketData creator = ClientBoundPacketData.create(PEPacketIDs.NETWORK_CHUNK_PUBLISHER_UPDATE_PACKET);
 				VarNumberSerializer.writeSVarInt(networkChunkUpdate, x);
 				VarNumberSerializer.writeVarInt(networkChunkUpdate, y);
 				VarNumberSerializer.writeSVarInt(networkChunkUpdate, z);
 				VarNumberSerializer.writeVarInt(networkChunkUpdate, view << 4);
 				return creator;
}
```
Then inside Chunk you just call:
```java
if (version.isAfterOrEq(ProtocolVersion.MINECRAFT_PE_1_8)) {
 	packets.add(createChunkPublisherUpdate(
        (int) movecache.getPEClientX(),
        (int) movecache.getPEClientX(),
        (int) movecache.getPEClientX(),
        Bukkit.getViewDistance() //TODO: client view distance yo
    ));
}
```
And inside position you call:
```java
if (version.isAfterOrEq(ProtocolVersion.MINECRAFT_PE_1_8)) {
 	packets.add(Chunk.createChunkPublisherUpdate(
        (int) x, (int) y, (int) z, Bukkit.getViewDistance() //TODO: client view distance yo
    ));
}
```

Reply:
damn forgot to push my last commits, should be... pretty much like this